daffodil-runtime1/src/main/scala/edu/illinois/ncsa/daffodil/dsom/CompiledExpression1.scala

```diff
@@ -386,6 +392,9 @@ class DPathElementCompileInfo(
         matchesERD
       }
 
+    retryMatchesERD.foreach { info =>
```

Comment:
Add comment why this is done for all the matches. It's because expressions can refer to multiple elements that have the same name, but are separate children of the element. A path expression that ends in such a path step is ambiguous, but ALL the elements of that name are being referenced by the expression.